### PR TITLE
TextColor - prevent unnecessary syscalls

### PR DESCRIPTION
**Pull request by pavelmash**
For FPC+Linux every call to write() =  syscall write

Since every syscall is: 
 - atomic
 - expensive

after this patch console coloring become a little bit faster and atomic (calling TextColor from different thread do not brake a terminal commands on 2 part)

(verified using `strace`)

### CODE REVIEW

File: SynCommons.pas [hunk 1]
```diff
@@ -55886,10 +55886,8 @@ procedure TextColor(Color: TConsoleColor);
     exit;
   TextAttr := ord(color);
   if ord(color)>=8 then
-    write(#27'[1;3') else
-    write(#27'[0;3');
-  write(AnsiTbl[(ord(color) and 7)+1],'m');
-  ioresult;
```
Comment (synopse) [inline SynCommons.pas]: why did you remove ioresult call ? IIRC it doesn't call any sys API.

Comment (pavelmash) [reply SynCommons.pas]: This is my mistake. I think call to ioresult it do nothing in this case, but now I understand it set global `InOutRes` variable to 0.
